backend/dataall/modules/dataset_sharing/services/share_managers/lf_share_manager.py

```diff
@@ -62,11 +61,7 @@ def get_share_principals(self) -> [str]:
         -------
         List of principals
         """
-        principal_iam_role_arn = IAM.get_role_arn_by_name(
-            account_id=self.target_environment.AwsAccountId,
-            role_name=self.share.principalIAMRoleName
-        )
-        principals = [principal_iam_role_arn]
+        principals = [f"arn:aws:iam::{self.target_environment.AwsAccountId}:role/{self.share.principalIAMRoleName}"]
```

Comment:
the former function was implicitly checking if the role exists - it would be good to keep this check

Reply:
This was edited due to merge conflicts. Added it back.